### PR TITLE
[macOS] Fixes unicode input with IME input mode inactive.

### LINKED ISSUE #28224: macOS: InputEventKey's unicode property is always 0
**Godot version:**
3.1

**OS/device including version:**
macOS Mojave (10.14.4)

**Issue description:**
On macOS, InputEventKey's unicode is always 0 unless a built-in control that supports text input (such as LineEdit) has the focus. This is an issue when building your own keyboard input controls. On Windows, the unicode property works as expected (i.e. it's non-zero for any key combination that has a unicode representation).

**Steps to reproduce:**
1. Create a scene with a Control in it
2. Add any input handler (_input, _unhandled_input, gui_input)
3. Check the value of the unicode property when a key such as 'A' is pressed

**Minimal reproduction project:**
[KeyboardInputTest.zip](https://github.com/godotengine/godot/files/3099773/KeyboardInputTest.zip)

### PR DESCRIPTION
**Pull request by bruvzg**
Fixes unicode input with IME input mode inactive.
Adds some IME input mode documentation and code comments.

Fixes #28224

### CODE REVIEW

File: doc/classes/InputEventKey.xml [hunk 1]
```diff
@@ -29,7 +29,7 @@
 			Key scancode, one of the [enum KeyList] constants.
 		</member>
 		<member name="unicode" type="int" setter="set_unicode" getter="get_unicode">
-			Key unicode identifier when relevant.
+			Key unicode identifier when relevant. Unicode identifiers for the composite characters and complex scripts may not be available unless IME input mode is active.
```
Comment (akien-mga) [inline doc/classes/InputEventKey.xml]: Maybe add a reference to `[method OS.set_ime_active]`?

File: doc/classes/OS.xml [hunk 2]
```diff
@@ -710,6 +712,9 @@
 			</argument>
 			<description>
 				Sets whether IME input mode should be enabled.
+				If active IME handles key events before the application and creates an composition string and suggestion list.
+				Application can retrieve the composition status by using [code]get_ime_selection[/code] and [code]get_ime_text[/code] functions.
```
Comment (akien-mga) [inline doc/classes/OS.xml]: Use `[method method_name]` instead of `[code]` to generate hyperlinks to methods.

Comment (bruvzg) [reply doc/classes/OS.xml]: Done.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**